java-strings/src/test/java/com/baeldung/string/SubstringPalindromeUnitTest.java

```diff
@@ -0,0 +1,66 @@
+package com.baeldung.string;
+
+import static org.junit.Assert.assertEquals;
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.Test;
+
+public class SubstringPalindromeUnitTest {
+
+    private static final String INPUT_NITIN = "nitin";
+    private static final String INPUT_CIVIC = "civic";
+    private static final String INPUT_LEVEL = "level";
+
```

Comment:
These three inputs are pretty much the same: they're all five letters, the entire word is a palindrome, and the three letters in the middle form a palindrome. I don't see any reason to show all three. Choose one ("civic"). Then, let's include different kinds of inputs. Here are some examples of things to test to prove the algorithms are correct:
1. a string that contains one or more non-trivial palindromes, but is not itself a palindrome, such as "ababac"
2. a string that contains a palindrome of even length, such as "indeed" -- this also covers an edge case, where a string ends with a palindrome
3. a string that begins with a palindrome, such as "bubble"

Reply:
Added different test cases